NewHaven/src/VGMap.cpp

```diff
@@ -272,29 +336,24 @@ Circle::~Circle(){
     delete isVisited;
     delete position;
     delete isPlayed;
-    delete building;
 }
 
 // override assignment operator
 Circle & Circle::operator=(const Circle &circle){
     if (this == &circle)
         return *this;
     else {
-        delete row;
-        delete column;
-        delete vCost;
-        delete isVisited;
-        delete isPlayed;
-        delete building;
-        row = new int(*circle.row);
-        column = new int(*circle.column);
-        vCost = new int(*circle.vCost);
-        isVisited = new bool(*circle.isVisited);
-        isPlayed = new bool(*circle.isPlayed);
-        // for some reason copy constructor of building is causing a segFault here
-        building = new Building();
-        *building = *circle.building;
-        position = new int(*circle.position);
+        *row = *circle.row;
+        *column = *circle.column;
+        *vCost = *circle.vCost;
+        *isVisited = *circle.isVisited;
+        *isPlayed = *circle.isPlayed;
+       if(circle.building != nullptr && building == nullptr)
+           // we dont create copies of buildings. We will clear it with the decks
+        building = circle.building;
+       else if(circle.building != nullptr && building != nullptr)
+           *building = *circle.building;
```

Comment:
`*building = *circle.building;` is an assignment by value so *circle.building will be implicitly copied to *building.
However, we don't want to make copies. Since we're not trying to copy, you can shallow copy the pointers. The good thing with that is that you don't need to check for nullptr anymore.
`building = circle.building` is valid even for nullptr. I told you that you were missing a check before because you were making a copy with a copy constructor, there a nullptr would have crashed it. In the case we have here, just do the shallow pointer assignment without any check.